yarn-project/sequencer-client/src/block_builder/circuit_block_builder.ts

```diff
@@ -168,13 +186,22 @@ export class CircuitBlockBuilder implements BlockBuilder {
     return new AppendOnlyTreeSnapshot(Fr.fromBuffer(treeInfo.root), Number(treeInfo.size));
   }
 
-  protected async runCircuits(txs: ProcessedTx[]): Promise<[RootRollupPublicInputs, Proof]> {
+  protected async runCircuits(txs: ProcessedTx[], newL1ToL2Messages: Fr[]): Promise<[RootRollupPublicInputs, Proof]> {
     // Check that the length of the array of txs is a power of two
     // See https://graphics.stanford.edu/~seander/bithacks.html#DetermineIfPowerOf2
     if (txs.length < 4 || (txs.length & (txs.length - 1)) !== 0) {
       throw new Error(`Length of txs for the block should be a power of two and at least four (got ${txs.length})`);
     }
 
+    // Check that the number of new L1 to L2 messages is the same as the max
+    // TODO: optionally check that they are all inserted in order, i.e. filled values at the front and empty at the end
+    // TODO: negative test for this ase
```

Comment:
Remember the todos.

Reply:
these have either been completed or made into tickets on the fairies board